### ISSUE TITLE
'Programming exercises:' Error when trying to edit the repository in the online IDE after the due date

### ISSUE THREAD

**Issue by laadvo** [ls1intum/Artemis#6124]
### Describe the bug

A student that has participated in a programming exercise is able to edit their repository after the submission due date in the online IDE. The changes cannot be saved or refreshed, since the two buttons are disabled, but after changing some lines of code the errors "One file or more could not be updated" and "You are not authorized to acess this page" appear.
When exiting the online IDE the changes are not saved, but the error "You are not authorized to acess this page" appears.

### To Reproduce

1. Paricipate in a programming exercise with the online IDE enabled before the exercise due date
2. After the due date open the IDE and start changing some lines of code
3. The two errors mentioned appear
4. Continue editing the files and the errors should appear again after some time
5. Exit the online IDE with changes that will be disregarded
6. "You are not authorized to acess this page" appears in the exercise page

### Expected behavior

After the due date the students should not be able to edit the repository, or at least the errors should not appear when doing so.
Since it says that the repository is locked it would make sense for the students to only be able to read their code without the ability to edit it.

### Screenshots

Buttons disabled and the repository being locked after the due date:
![image](https://user-images.githubusercontent.com/73339358/213865158-bd8cc8f9-8c95-4cef-8def-abc87c972e7f.png)

In the online IDE after editing some lines of code:
![image](https://user-images.githubusercontent.com/73339358/213864787-fbc1e32f-fdd5-4661-881a-6377e84f6fb6.png)

After leaving the online IDE, in the exercise page (the changes were not saved):
![image](https://user-images.githubusercontent.com/73339358/213864821-4e5d8b12-7907-4c6f-9959-d55882dabdda.png)

### Which version of Artemis are you seeing the problem on?

6.0.0

### What browsers are you seeing the problem on?

Chrome

**Comment by krusche** [comment]
I agree. In this case we should display the code editor in a read-only mode. Good catch!
